commit 0e622e80191e75c99b6ecc265c140a37d81e7a63
Author: Sricharan Ramabadhran <quic_srichara@quicinc.com>
Date:   Mon Jun 4 13:30:35 2018 -0700

    remoteproc: qcom: mdt_loader: Make the firmware authentication optional
    
    qcom_mdt_load function loads the mdt type firmware and
    initialises the secure memory as well. Make the initialisation only
    when requested by the caller, so that the function can be used
    by self-authenticating remoteproc as well.
    
    Acked-by: Andy Gross <andy.gross@linaro.org>
    Signed-off-by: Sricharan R <sricharan@codeaurora.org>
    Signed-off-by: Bjorn Andersson <bjorn.andersson@linaro.org>

diff --git a/include/linux/soc/qcom/mdt_loader.h b/include/linux/soc/qcom/mdt_loader.h
index 5b98bbdabc25..944b06aefb0f 100644
--- a/include/linux/soc/qcom/mdt_loader.h
+++ b/include/linux/soc/qcom/mdt_loader.h
@@ -17,4 +17,8 @@ int qcom_mdt_load(struct device *dev, const struct firmware *fw,
 		  phys_addr_t mem_phys, size_t mem_size,
 		  phys_addr_t *reloc_base);
 
+int qcom_mdt_load_no_init(struct device *dev, const struct firmware *fw,
+			  const char *fw_name, int pas_id, void *mem_region,
+			  phys_addr_t mem_phys, size_t mem_size,
+			  phys_addr_t *reloc_base);
 #endif